### PR TITLE
docs: Add package scope notices to READMEs \+ misc. README improvements

### PR DESCRIPTION
**Pull request by Josmithr**
Follow-up to #18223 and #18208.

Updates various library package READMEs to use the new pragmas to insert usage notices based on the intended package scope.

Also makes some other misc. README improvements.

### CODE REVIEW

File: experimental/framework/last-edited/README.md [hunk 1]
```diff
@@ -26,13 +35,13 @@ The update should always be called in response to a remote op because:
 
 The details returned in getLastEditDetails contain the `IUser` object and the `timestamp` of the last edit.
 
-# Last Edited Tracker Data Store
+## Last Edited Tracker Data Store
```
Comment (Josmithr) [inline experimental/framework/last-edited/README.md:38]: Fixed inconsistent heading levels. Everything should be parented under the same top-level heading.

File: experimental/framework/react-inputs/README.md
Comment (alexvy86) [inline experimental/framework/react-inputs/README.md:1]: The new content in this one seems incorrect

File: experimental/dds/tree2/README.md
Comment (alexvy86) [inline experimental/dds/tree2/README.md:1]: The new content is the one for tree instead of tree2?

Comment (Josmithr) [reply experimental/dds/tree2/README.md:1]: Urg. VSCode find and replace strikes again. There is a bug that results in whole-file replacement when trying to replace a subsection 😕

Will fix

File: experimental/dds/tree2/README.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
-# @fluid-experimental/tree2
+# @fluid-experimental/tree
```
Comment (Josmithr) [inline experimental/dds/tree2/README.md:1]:
```suggestion
# @fluid-experimental/tree2
```

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**